### ISSUE TITLE
`ICoreWebView2EnvironmentOptions4::SetCustomSchemeRegistrations` is wrong

### ISSUE THREAD

**Issue by aznhe21** [wravery/webview2-rs#18]
[`ICoreWebView2EnvironmentOptions4::SetCustomSchemeRegistrations`] has the following (wrong) signature:

```rust
    pub unsafe fn SetCustomSchemeRegistrations(
        &self,
        count: u32,
    ) -> ::windows::core::Result<ICoreWebView2CustomSchemeRegistration>
```

The following signature is correct:

```rust
    pub unsafe fn SetCustomSchemeRegistrations(
        &self,
        count: u32,
        schemeregistrations: *mut ICoreWebView2CustomSchemeRegistration
    ) -> ::windows::core::Result<()>
```

This is the same for [`ICoreWebView2EnvironmentOptions4_Impl`].

[`ICoreWebView2EnvironmentOptions4::SetCustomSchemeRegistrations`]: https://docs.rs/webview2-com/0.23.0/webview2_com/Microsoft/Web/WebView2/Win32/struct.ICoreWebView2EnvironmentOptions4.html#method.SetCustomSchemeRegistrations
[`ICoreWebView2EnvironmentOptions4_Impl`]: https://docs.rs/webview2-com/0.23.0/webview2_com/Microsoft/Web/WebView2/Win32/trait.ICoreWebView2EnvironmentOptions4_Impl.html

**Comment by wravery** [comment]
Yes, I noticed that too in #19. I am not sure if this can be fixed without a change to the WebView2 header to add out-param annotations to the method. For now, I worked around it by re-declaring the interface with the same `GUID` and correct signature using the `windows-interface` crate/macro, and that's the version of the interface I implemented.

I also added Rust APIs to these objects, so callers from Rust should not need to use the COM interfaces directly anymore. The re-declared interface is public, so they could use it, but they shouldn't need to.
